src/components/modal/types.ts

```diff
@@ -0,0 +1,23 @@
+import { ReactNode } from 'react';
+
+export interface ModalProps {
+    isOpen: boolean;
+    title: string | null;
+    size: WidthModal;
+    contentVariant?: ContentVariant | null;
+    onActionPrimary: () => void;
+    onActionSecondary: () => void;
+    buttonLabelPrimary: string;
+    buttonLabelSecondary: string;
+    children: ReactNode | ReactNode[];
+}
+
+export enum WidthModal {
+    SMALL = 'md:max-w-[400px]',
+    MEDIUM = 'md:max-w-[600px]',
```

Comment:
Maybe use S = 's' and M = 'm' here? And put the CSS to the classes in modal?
Also, if you want you can call it ModalWith oder ModalSize